Yoast/wordpress-seo
# Implement replacevar labels (nicenames)

## Summary

This PR can be summarized in the following changelog entry:

* In the snippet editor, replacement variables now have names that are more descriptive.

## Relevant technical choices:

* I've (for now) decided against using the `-` (or any of the other separators) or `Page x of y`, as I felt this goes against the rest of the labels. In essence, we already have the SnippetPreview to reflect the contents of the replacevars, and the SEO Title editorfield should just be clean and readable:
![image](https://user-images.githubusercontent.com/26220788/41040138-d5bec3b8-699b-11e8-9f6a-79b5b108d100.png)

The issue with having the labels reflecting the eventual look (for example not `Separator` but `*` `|` `~` `>>` etc), will (in the opinion of some of the team) not look as nice/intuitive when searching:

![image](https://user-images.githubusercontent.com/26220788/41040087-b4f57528-699b-11e8-8b2a-6205ef427a9e.png)

(Note that the dropdown menu does not contain the nicenames yet, will be fixed in https://github.com/Yoast/yoast-components/issues/577

Furthermore, I figured it would be a weird experience to write `%%Separator%%` and only see a blue `-` appear in the input field.

*NOTE:* These labels are easily changed, so if @jdevalk , @jono-alderson or @omarreiss feel differently about this it'll only be a bit of work.

## Test instructions

This PR can be tested by following these steps:

* Add some replacevars to the SnippetEditor fields. Update the post, and see that they now have nicenames. When you use the autocomplete, the nicenames will not immediately appear, this will be fixed in https://github.com/Yoast/yoast-components/issues/577
* Type (so do not autocomplete) a replacevar in full. So something like `%%title%%`. Replacevars entered that way will immediately be replaced by the nicenames.

## Quality assurance


Fixes #9835
